src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketTagSCMHead.java

```diff
@@ -39,15 +37,6 @@
 
     private static final long serialVersionUID = 1L;
 
-    /**
-     * Cache of the repository type.
-     *
-     * @since 2.2.11
-     */
-    // The repository type should be immutable for any SCMSource.
-    @CheckForNull
-    private final BitbucketRepositoryType repositoryType;
```

Comment:
Seems to be ok: https://stackoverflow.com/questions/16261383/delete-field-from-old-java-class-implementing-serializable/16261562

Reply:
Fine then. If a user installs a new version of the plugin and the results are serialised without `repositoryType`, and the user then downgrades the plugin, the field will be left null but that should be OK because it has `@CheckForNull`.